product-service/src/functions/catalogBatchProcess/handler.ts

```diff
@@ -0,0 +1,114 @@
+import 'source-map-support/register';
+import { Client } from 'pg';
+import { formatJSONResponse } from '@libs/apiGateway';
+import { middyfy } from '@libs/lambda';
+import AWS from 'aws-sdk';
+
+import { dbOptions } from '../../config/database-config';
+import { findProductByTitle, updateProduct, insertProduct } from '../../services/index';
+
+const catalogBatchProcess = async (event) => {
+    console.log("catalogBatchProcess lambda launched");
+    console.log(event);
+    const clientParams = {region: 'eu-west-1'};
+
+    try{      
+      AWS.config.update(clientParams);
+      const sns = new AWS.SNS();
+
+      const products = event.Records.map( ({body}) => JSON.parse(body) );
+
+      event.Records.forEach( (record) => {
+        console.log('messageAttributes', record.messageAttributes);
+      } );
+
+    // -- CONNECT TO DATABASE
+    let client;
+    if(process.env.NODE_ENV !== 'test'){
+      client = new Client(dbOptions);
+      await client.connect();
+    }else{
+      client = undefined;
+    }
+
+      for(const product of products){
+
+          console.log('product', product);
+
+          // const {title, description, price, count} = JSON.parse(product);
+          const {title} = product;
+
+          const foundProduct = await findProductByTitle(client, title);
+
+          console.log('check product from repository with title: ' + title + ', result = ', foundProduct);
+
+          if(foundProduct){
+            console.log('update product by id', foundProduct.id);
+
+            await updateProduct(client, product, foundProduct.id);
+
+              // // -- BEGIN TRANSACTION
```

Comment:
I also advise you to remove the commented code. If you need to make a memo on some approach/code it is much better and more convenient to use `git stash`. Finding these changes is as easy as any commit or branch in the repository and keeps the code clean.